system/Boot.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter;
+
+use CodeIgniter\Config\DotEnv;
+use CodeIgniter\Exceptions\FrameworkException;
+use Config\Autoload;
+use Config\Modules;
+use Config\Paths;
+use Config\Services;
+
+/**
+ * Bootstrap for the application
+ *
+ * @codeCoverageIgnore
+ */
+class Boot
+{
+    /**
+     * @used-by public/index.php
+     *
+     * Context
+     *   web:     Invoked by HTTP request
+     *   php-cli: Invoked by CLI via `php public/index.php`
+     */
+    public static function bootWeb(Paths $paths): void
+    {
+        static::loadDotEnv($paths);
+        static::defineEnvironment();
+        static::loadEnvironmentBootstrap($paths);
+        static::definePathConstants($paths);
+        if (! defined('APP_NAMESPACE')) {
+            static::loadConstants();
+        }
+        static::loadCommonFunctions();
+        static::loadAutoloader();
+        static::setExceptionHandler();
+        static::checkMissingExtensions();
+        static::initializeKint();
+    }
+
+    /**
+     * @used-by spark
+     */
+    public static function bootSpark(Paths $paths): void
+    {
+        static::loadDotEnv($paths);
+        static::defineEnvironment();
+        static::loadEnvironmentBootstrap($paths);
+        static::definePathConstants($paths);
+        if (! defined('APP_NAMESPACE')) {
+            static::loadConstants();
+        }
+        static::loadCommonFunctions();
+        static::loadAutoloader();
+        static::setExceptionHandler();
+        static::checkMissingExtensions();
+        static::initializeKint();
+    }
+
+    /**
+     * @used-by system/Test/bootstrap.php
+     */
+    public static function bootTest(Paths $paths): void
+    {
+        static::loadDotEnv($paths);
+        static::loadEnvironmentBootstrap($paths, false);
+        static::loadConstants();
+        static::loadCommonFunctions();
+        static::loadAutoloader();
+        static::setExceptionHandler();
+        static::checkMissingExtensions();
```

Comment:
This feels awfully low down the stack. We really don't need these extensions in any of the prior methods? I recognize that the `FrameworkException` relies on Autoloader and probably Exception Handling, but we could convert it to a simpler echo + exit similar to the environment error.

Reply:
It means we stop providing the translated error message.

Reply:
Changed the method order, and changed it to echo + exit.